### ISSUE TITLE
Write unittests

### ISSUE THREAD

**Issue by vpetersson** [databat-io/sonar#22]
Some of the code is hard to test, but we should at least get *some* tests going (and hook it up with CI).

**Comment by saarmstrong** [comment]
Are you looking for volunteers?

**Comment by sulrich70** [comment]
this seems to be more are more useful with health restrictions for covid measures. Anyone interested in continuing this?

**Comment by vpetersson** [comment]
@saarmstrong 
> Are you looking for volunteers?

Yeah sure -- i haven't had much bandwidth to this myself in recent time.


@sulrich70 
> this seems to be more are more useful with health restrictions for covid measures. Anyone interested in continuing this?

Yeah sure -- feel free to get at it and add contributions.
